python/setup.py

```diff
@@ -270,23 +235,28 @@ def append_cmake_bool(value, varname):
                 cmake_options.append('-D{0}={1}'.format(
                     varname, 'on' if value else 'off'))
 
+            def maybe_append_cmake_bool(varname):
+                env_var = varname.replace("PYARROW_BUILD", "PYARROW_WITH")
+                if env_var in os.environ:
+                    value = strtobool(os.environ.get(env_var))
+                    append_cmake_bool(value, varname)
```

Comment:
How about doing this in `CMakeLists.txt` too?
We can refer an environment variable by `"$ENV{PYARROW_WITH_CUDA}` in `CMakeLists.txt`.
See also: https://cmake.org/cmake/help/latest/variable/ENV.html

Reply:
I did that initially in a first version locally, but then moved it again to setup.py to avoid the same problem with it being cached. But with you suggestion above to use an AUTO default, then I can indeed move it back to the cmake.